terraform-aws-static-site/main.tf

```diff
@@ -0,0 +1,161 @@
+provider "aws" {
+    region = "us-west-2"
+}
+
+# Non www-bucket
+resource "aws_s3_bucket" "b" {
+  bucket = var.bucket_name
+  acl    = "public-read"
+
+  website {
+    index_document = "index.html"
+    error_document = "index.html"
+  }
+}
+
+data "aws_iam_policy_document" "b" {
+  statement {
+    sid    = "PublicReadGetObject"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    actions   = ["s3:GetObject"]
+    resources = ["arn:aws:s3:::${var.bucket_name}/*"]
+  }
+}
+
+resource "aws_s3_bucket_policy" "b" {
+  bucket = aws_s3_bucket.b.id
+  policy = data.aws_iam_policy_document.b.json
+}
+
+# www-bucket
+resource "aws_s3_bucket" "www-bucket" {
+  bucket = "www-${var.www_bucket_name}"
+  acl    = "public-read"
+
+  website {
+    redirect_all_requests_to = "https://${var.redirect_address}"
+  }
+}
+
+# Non www-bucket CDN
+resource "aws_cloudfront_distribution" "s3_distribution" {
+  origin {
+    domain_name = var.domain_name
+    origin_id   = var.origin_id
+
+    custom_origin_config {
+      origin_protocol_policy = "http-only"
+      http_port              = "80"
+      https_port             = "443"
+      origin_ssl_protocols   = ["TLSv1"]
+    }
+  }
+
+  price_class = "PriceClass_100"
+
+  enabled = true
+
+  aliases = var.aliases
+
+  default_cache_behavior {
+    allowed_methods  = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+    cached_methods   = ["GET", "HEAD"]
+    target_origin_id = var.origin_id
+
+    forwarded_values {
+      query_string = true
+
+      cookies {
+        forward = "none"
+      }
+    }
+
+    viewer_protocol_policy = "redirect-to-https"
+    min_ttl                = 0
+    default_ttl            = 3600
+    max_ttl                = 86400
+  }
+
+  viewer_certificate {
+    cloudfront_default_certificate = true
+  }
+
+  restrictions {
+    geo_restriction {
+      restriction_type = "none"
+    }
+  }
+
+  custom_error_response {
+    error_caching_min_ttl = "0"
+    error_code            = "403"
+    response_code         = "200"
+    response_page_path    = "/index.html"
+  }
+}
+
+// www-bucket CDN
+resource "aws_cloudfront_distribution" "www_s3_distribution" {
+  origin {
+    domain_name = var.www_domain_name
```

Comment:
We can actually do this without variables in a much easier way! Terraform resources have output values. Things that are known after the creation of a resource. So after you create a bucket, you will know the domain name it exists under, check out the docs here =>

Attribute reference: https://www.terraform.io/docs/providers/aws/r/s3_bucket.html#attributes-reference
Resource reference (shows an example of resource dependencies): https://www.terraform.io/docs/configuration/resources.html

Have a go and see if you can remove the need for `var.www_domain_name` and `var.domain_name`

Reply:
I think I did it properly. It works on the local side of things once I updated the autotelic-site/infrastructure branch